coreTable/OwidTable.ts

```diff
@@ -615,6 +631,154 @@ export class OwidTable extends CoreTable<OwidRow, OwidColumnDef> {
         )
     }
 
+    interpolateColumnsByClosestTimeMatch(
```

Comment:
This fn is huge, and I'm sure there are ways to break it up, but seems like it is just a complex set of operations and more importantly the tests look good so we can always break it up later.